pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -25,21 +27,52 @@ stdenv.mkDerivation rec {
 
   sourceRoot = ".";
 
-  patches = lib.optional enableNixHacks ./nix-hacks.patch;
+  patches = lib.optionals (stdenv.hostPlatform.isDarwin) [ ./darwin.patch ] ++ lib.optionals (enableNixHacks) [ ./nix-hacks.patch ];
 
   # Bazel expects several utils to be available in Bash even without PATH. Hence this hack.
 
-  customBash = writeScriptBin "bash" ''
-    #!${stdenv.shell}
-    PATH="$PATH:${lib.makeBinPath [ coreutils ]}" exec ${bash}/bin/bash "$@"
+  customBash = writeCBin "bash" ''
+    #include <stdio.h>
+    #include <stdlib.h>
+    #include <string.h>
+    #include <unistd.h>
+
+    extern char **environ;
+
+    int main(int argc, char *argv[]) {
+      printf("environ: %s\n", environ[0]);
+      char *path = getenv("PATH");
+      char *pathToAppend = "${lib.makeBinPath [ coreutils ]}";
+      char *newPath;
+      if (path != NULL) {
+        int length = strlen(path) + 1 + strlen(pathToAppend) + 1;
+        newPath = malloc(length * sizeof(char));
+        snprintf(newPath, length, "%s:%s", path, pathToAppend);
+      } else {
+        newPath = pathToAppend;
+      }
+      setenv("PATH", newPath, 1);
+      execve("${bash}/bin/bash", argv, environ);
+      return 0;
+    }
   '';
 
-  postPatch = ''
+  postPatch = stdenv.lib.optionalString stdenv.hostPlatform.isDarwin ''
+    export NIX_LDFLAGS="$NIX_LDFLAGS -F${CoreFoundation}/Library/Frameworks -F${CoreServices}/Library/Frameworks -F${Foundation}/Library/Frameworks"
```

Comment:
Ideally we wouldn't need to specify these explicitly, as https://github.com/NixOS/nixpkgs/blob/bbcaf78350d87c4657b1a2451132535467a454eb/pkgs/build-support/bintools-wrapper/setup-hook.sh should add these automatically, like it does `-L` flags, and like how https://github.com/NixOS/nixpkgs/blob/bbcaf78350d87c4657b1a2451132535467a454eb/pkgs/build-support/cc-wrapper/setup-hook.sh adds `-F` flags, but when I tried to fix that I found out a bunch of packages are finicky on macOS so I gave up.

Reply:
Tried it:

```
ERROR: /private/tmp/nix-build-bazel-0.12.0.drv-0/third_party/protobuf/3.4.0/BUILD:399:1: Linking of rule '//third_party/protobuf/3.4.0:js_embed' failed (Exit 1): cc_wrapper.sh failed: error executing command
  (cd /tmp/bazel_FhkpSXM4/out/execroot/io_bazel && \
  exec env - \
    APPLE_SDK_PLATFORM='' \
    APPLE_SDK_VERSION_OVERRIDE='' \
    PATH=/nix/store/allybvvm28bybqrjpkdqvqi7d4r5x38q-zip-3.0/bin:/nix/store/10f4qxb00wk9j6cicd9h1qzwqgnc36ji-python-2.7.14/bin:/nix/store/dparbm2kq903rx6kp1m9zy1l2dcsxcmx-unzip-6.0/bin:/nix/store/gkzi1rn6qlqnalynjifmzgfzwh0x5aa2-which-2.21/bin:/nix/store/468wzkb2swaj4zdc1p6biiw1157jn673-bash/bin:/nix/store/8b0gj3bmcy5bh1zcnhn0146199b3m5kb-clang-wrapper-5.0.1/bin:/nix/store/p85m242dg1zd85ln3cg0agx0q1n0n6h0-clang-5.0.1/bin:/nix/store/5lkrw9dnsgy62qm1ampvww1c5n1pdm4b-coreutils-8.29/bin:/nix/store/jxmp27m26646f67faji9v94d5471ma8y-cctools-binutils-darwin-wrapper/bin:/nix/store/kdff2gim6417493yha769kh00n63lnrw-cctools-binutils-darwin/bin:/nix/store/5lkrw9dnsgy62qm1ampvww1c5n1pdm4b-coreutils-8.29/bin:/nix/store/bc5fk65ijws2938z3nlvkay7i83qjvp7-zulu1.8.0_121-8.20.0.5/bin:/nix/store/5lkrw9dnsgy62qm1ampvww1c5n1pdm4b-coreutils-8.29/bin:/nix/store/lzrwadp3l7np2734jxzqmxsgnjqfy4m7-findutils-4.6.0/bin:/nix/store/jrcykkx415grrjvrrhvs8hgl6pvj6x5b-diffutils-3.6/bin:/nix/store/llk6iyb3ppvyaflbkg5rbv960wlsxpkc-gnused-4.4/bin:/nix/store/c3sa761qjcaa5rg0s8h8qxj0mghkidv5-gnugrep-3.1/bin:/nix/store/gpp9x3ninvcw6m2n5m075q8nckzmcv09-gawk-4.2.1/bin:/nix/store/m88jhn3qgiz48zdllhpvgkz1ricmh89m-gnutar-1.30/bin:/nix/store/61bqs4awrycwdiaz7yb3bxvbyja1ghxr-gzip-1.9/bin:/nix/store/7vldi2061ga9phyhkr3sin8fk2kpc6sl-bzip2-1.0.6.0.1-bin/bin:/nix/store/y2nxw92is8125gdng3bsv23isray1g9m-gnumake-4.2.1/bin:/nix/store/r8bx3qf1bpncb14i9gzma4vr089pc3pv-bash-4.4-p19/bin:/nix/store/l1d4iaik29pkg7pdng7iw8ch75h71n36-patch-2.7.6/bin:/nix/store/03x40jq9ik0j4br0k2jmn288hz25589l-xz-5.2.3-bin/bin:/nix/store/5lkrw9dnsgy62qm1ampvww1c5n1pdm4b-coreutils-8.29/bin \
    XCODE_VERSION_OVERRIDE=9.3.1 \
  external/local_config_cc/cc_wrapper.sh -fobjc-link-runtime -Wl,-S -o bazel-out/host/bin/third_party/protobuf/3.4.0/js_embed bazel-out/host/bin/third_party/protobuf/3.4.0/_objs/js_embed/third_party/protobuf/3.4.0/src/google/protobuf/compiler/js/embed.o -headerpad_max_install_names -lc++ -no-canonical-prefixes -Wl,-L/nix/store/10f4qxb00wk9j6cicd9h1qzwqgnc36ji-python-2.7.14/lib -Wl,-L/nix/store/zl3vsrqfwx4xvxagdrdqav73yl53cxps-libc++-5.0.1/lib -Wl,-L/nix/store/anpyw2fy2f4qfz3d5yy0x9yxfxnx7sxy-objc4-osx-10.11.6/lib -Wl,-L/nix/store/zl3vsrqfwx4xvxagdrdqav73yl53cxps-libc++-5.0.1/lib -Wl,-L/nix/store/bc5fk65ijws2938z3nlvkay7i83qjvp7-zulu1.8.0_121-8.20.0.5/lib -Wl,-L/nix/store/10f4qxb00wk9j6cicd9h1qzwqgnc36ji-python-2.7.14/lib -Wl,-L/nix/store/zl3vsrqfwx4xvxagdrdqav73yl53cxps-libc++-5.0.1/lib -Wl,-L/nix/store/anpyw2fy2f4qfz3d5yy0x9yxfxnx7sxy-objc4-osx-10.11.6/lib -Wl,-L/nix/store/zl3vsrqfwx4xvxagdrdqav73yl53cxps-libc++-5.0.1/lib -Wl,-L/nix/store/bc5fk65ijws2938z3nlvkay7i83qjvp7-zulu1.8.0_121-8.20.0.5/lib)

ld: framework not found Foundation
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Do you mean in normal operation `cc_wrapper` would take care of that?

Reply:
I need to specify it in the `postPatch` so I can use the updated flags to patch the build commands.